### ISSUE TITLE
General Security Issues

### ISSUE THREAD

**Issue by GoogleCodeExporter** [srulad/clients-oriented-ftp#266]
```
Upon looking at the code i see $_GET variables being used directly in sql 
queries and functions. I do not see any input validation or escaping. Is this 
correct?


What version of the product are you using? On what operating system?
r405

Please provide any additional information below.

I hope I am missing something here but if there truly is no input validation 
then the whole application is a security risk. :(
```

Original issue reported on code.google.com by `shawn.k...@gmail.com` on 19 Apr 2013 at 6:25

**Comment by GoogleCodeExporter** [comment]
```
Thanks Shawn.
Where are you seeing this queries?
```

Original comment by `i...@subwaydesign.com.ar` on 19 Apr 2013 at 8:26

**Comment by GoogleCodeExporter** [comment]
```
process.php is an example but there are many locations of this.

process.php:

line 44:  $this->sum_sql = 'UPDATE tbl_files_relations SET 
download_count=download_count+1 WHERE file_id="' . $_GET['id'] .'"'; (an 
attacker could use sql injection here to mess with the database. There are a 
lot of sql injection issues in your code)
line 85: $file = UPLOADED_FILES_FOLDER.$_GET['url']; (this one is really bad 
because an attacker could set url=../../../../../../../../../etc/passwd and 
theoretically get the passwd file)

Basically anywhere you use $_GET directly in code is bad unless you have an 
input validation function somewhere that parses the _GET array before you do 
any real work with it.

If you were to write a function that runs before the rest of your code that 
cleans up and validates _GET variables, then using _GET would be fine. I didn't 
see one in your code anywhere. :(
```

Original comment by `shawn.k...@gmail.com` on 19 Apr 2013 at 8:54

**Comment by GoogleCodeExporter** [comment]
```
I see your point. I didn't add any since there are are several checks before 
reaching any part of the code that uses the GET values. But I guess I should 
add them.
Thanks!
```

Original comment by `i...@subwaydesign.com.ar` on 19 Apr 2013 at 10:28
- Changed state: **Started**
